OpenOversight/app/main/views.py

```diff
@@ -801,6 +804,110 @@ def download_dept_csv(department_id):
     return Response(csv, mimetype="text/csv", headers=csv_headers)
 
 
+def check_output(output_str):
+    if output_str == "Not Sure":
+        return ""
+    return output_str
+
+
+@main.route('/download/department/<int:department_id>/officers', methods=['GET'])
+@limiter.limit('5/minute')
+def download_dept_officers_csv(department_id):
+    department = Department.query.filter_by(id=department_id).first()
+    if not department:
+        abort(404)
+
+    officers = (db.session.query(Officer)
+                .options(joinedload(Officer.assignments_lazy)
+                         .joinedload(Assignment.job)
+                         )
+                .options(joinedload(Officer.salaries))
+                .filter_by(department_id=department_id)
+                )
+
+    if not officers:
+        abort(404)
+    csv_output = io.StringIO()
+    csv_fieldnames = ["id", "unique identifier", "last name", "first name", "middle initial", "suffix", "gender",
+                      "race", "birth year", "employment date", "badge number", "job title", "most recent salary"]
+    csv_writer = csv.DictWriter(csv_output, fieldnames=csv_fieldnames)
+    csv_writer.writeheader()
+
+    for officer in officers:
+        if officer.assignments_lazy:
+            most_recent_assignment = max(officer.assignments_lazy, key=lambda a: a.star_date)
```

Comment:
Looks like this can cause a TypeError:

```
File "/usr/src/app/OpenOversight/app/main/views.py", line 838, in download_dept_officers_csv
most_recent_assignment = max(officer.assignments_lazy, key=lambda a: a.star_date)
TypeError: '>' not supported between instances of 'NoneType' and 'NoneType'
```

Reply:
Thanks! This should be fixed now. I also checked for other fields that could be `None` and would break things, hoping I didn't miss any this time